src/lib/genesis_ledger_helper/lib/genesis_ledger_helper_lib.ml

```diff
@@ -645,3 +624,193 @@ let runtime_config_of_precomputed_values (precomputed_values : Genesis_proof.t)
+      in
+      (* test field-by-field, to track down any errors *)
+      assert (
+        Public_key.Compressed.equal account.public_key test_account.public_key ) ;
+      assert (Token_id.equal account.token_id test_account.token_id) ;
+      assert (
+        Account.Token_symbol.equal account.token_symbol
+          test_account.token_symbol ) ;
+      assert (Account.Nonce.equal account.nonce test_account.nonce) ;
+      assert (
+        Receipt.Chain_hash.equal account.receipt_chain_hash
+          test_account.receipt_chain_hash ) ;
+      assert (Option.is_some account.delegate) ;
+      assert (
+        Option.equal Public_key.Compressed.equal account.delegate
+          test_account.delegate ) ;
+      assert (State_hash.equal account.voting_for test_account.voting_for) ;
+      assert (Account.Timing.equal account.timing test_account.timing) ;
+      assert (Permissions.equal account.permissions test_account.permissions) ;
+      assert (Option.equal Zkapp_account.equal account.zkapp test_account.zkapp)
+
+    (* if nondefault token, no delegate created from runtime config *)
+    let%test_unit "non-zkApp ledger, nondefault token" =
+      let runtime_accounts =
+        match non_zkapp_ledger_nondefault_token.base with
+        | Runtime_config.Ledger.Accounts accts ->
+            accts
+        | _ ->
+            failwith "Expected accounts in ledger"
+      in
+      let accounts =
+        Accounts.to_full runtime_accounts
+        |> List.map ~f:(fun (_sk, account) -> account)
+      in
+      assert (List.length accounts = 1) ;
+      let account = List.hd_exn accounts in
+      let test_account =
+        let account_id =
+          Mina_base.Account_id.create
+            (Public_key.Compressed.of_base58_check_exn pk)
+            nondefault_token
+        in
+        let balance = Currency.Balance.of_mina_int_exn 49_000_000 in
+        Mina_base.Account.create account_id balance
+      in
+      assert (Option.is_none test_account.delegate) ;
+      assert (Option.is_none account.delegate)
+
+    (* zkApp account fields in runtime config are all required, but we can make verification key `null` *)
+    let%test_unit "zkApp ledger" =
+      let runtime_accounts =
+        match zkapp_ledger.base with
+        | Runtime_config.Ledger.Accounts accts ->
+            accts
+        | _ ->
+            failwith "Expected accounts in ledger"
+      in
+      let accounts =
+        Accounts.to_full runtime_accounts
+        |> List.map ~f:(fun (_sk, account) -> account)
+      in
+      assert (List.length accounts = 1) ;
+      let account = List.hd_exn accounts in
+      let default = Mina_base.Zkapp_account.default in
+      let zkapp_account =
+        match account.zkapp with
+        | None ->
+            failwith "Expected zkApp account in account"
+        | Some zkapp ->
+            zkapp
+      in
+      assert (
+        Option.equal Verification_key_wire.equal zkapp_account.verification_key
```

Comment:
Do we check that the vk hash in the config is correct? ie., matches the hash of the corresponding vk

Reply:
There's no vk hash in the config. It's computed when converting to a `Zkapp_account.t` here: https://github.com/MinaProtocol/mina/blob/feature/runtime-config-test/src/lib/genesis_ledger_helper/lib/genesis_ledger_helper_lib.ml#L133